shared/util/quit-helper.desktop.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import electron from 'electron'
+import {ipcRenderer, ipcMain} from 'electron'
+import {quit} from '../../desktop/app/ctl'
+
+const app = electron.app || electron.remote.app
+const BrowserWindow = electron.BrowserWindow || electron.remote.BrowserWindow
+
+type WindowContext = {type: 'uiWindow'} | {type: 'menubar'} | {type: 'popup'} | {type: 'mainWindow'} | {type: 'mainThread'}
+
+export type Context = {type: 'mainThread'} | {type: 'quitButton'} | {type: 'unknown'} | WindowContext
+
+export type Action = {type: 'closePopups'} | {type: 'hideMainWindow'} | {type: 'quitApp'}
+
+// Logic to figure out what to do given your context
+export function quitOnContext (context: Context): Array<Action> {
```

Comment:
Would be nice to have this in /desktop since there isn't really a corresponding mobile impl that makes any sense... though I guess we don't get flow checks there?